features/git-town-prepend/on-grandchild-branch.feature

```diff
@@ -0,0 +1,71 @@
+      | existing-parent  | local and remote | existing_parent_commit  | existing_parent_file  | existing parent content  |
+    And I am on the "existing-feature" branch
+    And I have an uncommitted file
+
+
+  Scenario: inserting a branch into the branch ancestry
+    When I run `git town-prepend new-parent`
+    Then it runs the commands
+      | BRANCH           | COMMAND                                    |
+      | existing-feature | git fetch --prune                          |
+      |                  | git add -A                                 |
+      |                  | git stash                                  |
+      |                  | git checkout main                          |
+      | main             | git rebase origin/main                     |
+      |                  | git checkout existing-parent               |
+      | existing-parent  | git merge --no-edit origin/existing-parent |
+      |                  | git merge --no-edit main                   |
+      |                  | git checkout -b new-parent existing-parent |
+      | new-parent       | git push -u origin new-parent              |
+      |                  | git stash pop                              |
+    And I end up on the "new-parent" branch
+    And I still have my uncommitted file
+    And I have the following commits
+      | BRANCH           | LOCATION         | MESSAGE                 |
+      | existing-feature | local and remote | existing_feature_commit |
+      | existing-parent  | local and remote | existing_parent_commit  |
+      | new-parent       | local and remote | existing_parent_commit  |
+    And Git Town is now aware of this branch hierarchy
+      | BRANCH           | PARENT          |
+      | existing-feature | new-parent      |
+      | existing-parent  | main            |
+      | new-parent       | existing-parent |
+
+
+  Scenario: Undo
+    Given I run `git town-prepend new-parent`
+    When I run `git town-prepend --undo`
+    Then it runs the commands
+        | BRANCH           | COMMAND                       |
+        | new-parent       | git add -A                    |
+        |                  | git stash                     |
+        |                  | git push origin :new-parent   |
+        |                  | git checkout existing-parent  |
+        | existing-parent  | git branch -d new-parent      |
+        |                  | git checkout main             |
+        | main             | git checkout existing-feature |
+        | existing-feature | git stash pop                 |
+    And I end up on the "existing-feature" branch
+    And I still have my uncommitted file
+    And I am left with my original commits
+    And Git Town is now aware of this branch hierarchy
+      | BRANCH           | PARENT          |
+      | existing-feature | existing-parent |
+      | existing-parent  | main            |
```

Comment:
I would have suggested keeping the child test and removing the grandchild test, since they are equivalent to me and the child branch requires less setup. The child test shouldn't match git-hack as its prepending the branch instead of appending